src/lib/classes/styling/GlobalCssSchemesLoader.ts

```diff
@@ -0,0 +1,206 @@
+				this.applyCssVariables(scheme);
+			}
+		});
+	}
+
+	/**
+	 * Method for applying CSS variabels for a specific mediafeature
+	 * @param feature
+	 */
+	applyCssVariables(feature: z.infer<typeof MediaScheme>) {
+		// Apply color variables
+		if (feature.color) {
+			for (const [key, val] of Object.entries(feature.color)) {
+				this._window.document.documentElement.style.setProperty(
+					key,
+					this.createCssColor(val),
+				);
+				this._propertyNames.push(key);
+			}
+		}
+
+		// Apply font size variables
+		if (feature.fontSize) {
+			for (const [key, val] of Object.entries(feature.fontSize)) {
+				this._window.document.documentElement.style.setProperty(
+					key,
+					val[0] + val[1], //TODO: Check the font size number
+				);
+				this._propertyNames.push(key);
+			}
+		}
+
+		// Apply border variables
+		if (feature.border) {
+			for (const [key, val] of Object.entries(feature.border)) {
+				this._window.document.documentElement.style.setProperty(
+					key,
+					this.createCssColor(val[2]) /* Border color */ +
+						" " +
+						val[0] /* Border style */ +
+						" " +
+						val[1][0] /* Border size */ +
+						val[1][1] /* Border size unit */,
+				);
+				this._propertyNames.push(key);
+			}
+		}
+	}
+
+	/**
+	 * Method for clearing the applied styles
+	 */
+	private clearAppliedProperties() {
+		this._propertyNames.forEach((attribute) => {
+			this._window.document.documentElement.style.removeProperty(
+				attribute,
+			);
+		});
+
+		this._propertyNames = [];
+	}
+
+	/**
+	 * Method for re-applying the specified styles
+	 */
+	reapplyMediaFeatures() {
+		this.clearAppliedProperties();
+		this.applySchemes();
+	}
+
+	/**
+	 * Method for loading the GlobalCssProperties.json file
+	 */
+	private parseMediaFeatures(): z.infer<typeof MediaSchemes> {
+		// Parsing media features
+		const parsedMediaFeatures = MediaSchemes.safeParse(GlobalCssProperties);
+
+		// Throwing error if the parsing failed
+		if (!parsedMediaFeatures.success) {
+			throw new Error(parsedMediaFeatures.error.message);
+		}
+
+		return parsedMediaFeatures.data;
+	}
+
+	/**
+	 * Method for adding appropriate event listeners
+	 */
+	private addEventListeners() {
+		this._supportedMediaFeatures.forEach((feature) => {
+			this._window
+				.matchMedia(`(${feature})`)
+				.addEventListener("change", () => {
+					this.reapplyMediaFeatures();
+				});
+		});
+	}
+
+	/**
+	 * Method for checking and creating CSS color string
+	 * @param color
+	 * @returns CSS color string
+	 */
+	private createCssColor(color: z.infer<typeof ColorValue>): string {
+		const supportedGamuts: string[] = [
+			"srgb",
+			"srgb-linear",
+			"display-p3",
+			"a98-rgb",
+			"prophoto-rgb",
+			"rec2020",
+			"xyz",
+			"xyz-d50",
+			"xyz-d65",
+		];
+		let cssColor: string;
+
+		// Check if color gamut is supported
+		if (!supportedGamuts.includes(color[0])) {
+			throw new Error(
+				`Color gamut "${color[0]}" specified in parsed global css styles, is not supported."`,
+			);
```

Comment:
Why are you doing this check manually? You can define the Zod schema to only parse a [`union`](https://zod.dev/?id=unions) of [`literals`](https://zod.dev/?id=literals) that match `supportedGamuts`, and youre done.